python/test/unit/language/test_core.py

```diff
@@ -4804,18 +4804,29 @@ def test_convert2d(M, N, src_layout, interm_layout, dst_layout, dtype, device):
         pytest.xfail("Out of bound access when maxPhase > 1")
     if str(src_layout) == str(dst_layout):
         pytest.xfail("Do not convert same layout")
-    if is_hip():
+    if is_hip() or is_xpu():
         try:
             scratch_shape = compute_scratch_buffer_shape(src_layout, dst_layout, (M, N))
         except AssertionError:
             pytest.skip("Can't compute scratch buffer size")
-        lds_size = 65536
-        # consider int32 dtype in scratch buffer size,
-        # because it is the largest dtype used in convert_layout in this test
-        int32_size = 4
-        # skip even if scratch buffer equal to lds_size, because real scratch buffer is typically larger due to padding
-        if scratch_shape[0] * scratch_shape[1] * int32_size >= lds_size:
-            pytest.skip("Scratch buffer is too large")
+        if is_hip():
+            lds_size = 65536
+            # consider int32 dtype in scratch buffer size,
+            # because it is the largest dtype used in convert_layout in this test
+            int32_size = 4
+            # skip even if scratch buffer equal to lds_size, because real scratch buffer is typically larger due to padding
+            if scratch_shape[0] * scratch_shape[1] * int32_size >= lds_size:
+                pytest.skip("Scratch buffer is too large")
+        elif is_xpu():
+            scratch_shape = compute_scratch_buffer_shape(src_layout, dst_layout, (M, N))
+            shared_mem_size = triton.runtime.driver.active.utils.get_device_properties(
+                triton.runtime.driver.active.get_current_device())["max_shared_mem"]
+            # consider int32 dtype in scratch buffer size,
+            # because it is the largest dtype used in convert_layout in this test
+            int32_size = 4
+            # skip even if scratch buffer equal to shared memory size, because real scratch buffer is typically larger due to padding
+            if scratch_shape[0] * scratch_shape[1] * int32_size >= shared_mem_size:
+                pytest.skip("XPU: Scratch buffer is too large")
```

Comment:
I assume there is nothing we can do about that, so not a failure we should count toward pass rate calculation, let's change it to `xfail`.
```suggestion
                pytest.xfail("Scratch buffer is too large")
```

Reply:
I'm not sure yet. Perhaps we can modify the block/grid size parameters to make the test pass? But it looks like xfail is used in similar situations so I will switch to `xfail`.